src/test/Makefile

```diff
@@ -438,11 +441,28 @@ COMMON_FLAGS = \
 	-O0 \
 	-DUNIT_TEST \
 	-isystem $(GTEST_DIR)/inc \
-	-MMD -MP
+	-MMD -MP \
+	-Wno-c99-extensions \
+	-Wno-reorder
 
 CC_VERSION = $(shell $(CC) -dumpversion)
 CXX_VERSION = $(shell $(CXX) -dumpversion)
 
+# Please revisit versions when new clang version arrive.
+CC_VERSION_MAJOR := $(shell expr `$(CC) -dumpversion | cut -f1 -d.`)
+CC_VERSION_CHECK_MIN := 7
+# Add 1 to version requested
+CC_VERSION_CHECK_MAX := 11
+ifdef MACOSX
+CC_VERSION_CHECK_MAX := 13
+endif
+ifeq ($(shell expr $(CC_VERSION_MAJOR) \>= $(CC_VERSION_CHECK_MAX)),1)
+    $(error $(CC) $(CC_VERSION_MAJOR) is not supported. On most systems the correct compiler will be available as 'clang-10' and 'clang++-10, but Betaflight invokes 'clang' and 'clang++' to accommodate some operating systems that do not have proper clang compilers available. Please modify your system so that an appropriate version of clang is invoked as 'clang' and 'clang++'.)
+endif
+ifneq ($(shell expr $(CC_VERSION_MAJOR) \>= $(CC_VERSION_CHECK_MIN)),1)
```

Comment:
It will be easier to maintain if there is only one check, seeing that the error message is the same: `ifeq ($(shell expr $(CC_VERSION_MAJOR) \< $(CC_VERSION_CHECK_MIN) \| $(CC_VERSION_MAJOR) \> $(CC_VERSION_CHECK_MAX)),1)`.

Reply:
Waiting for Travis build. Locally this works! As all I did locally 😞. I was trying to maintain one check all the way. 
CC_VERSION_CHECK_WORKING_MAX 😆

Reply:
Thank you 1000 times. This equation just works also on Travis !!!

Reply:
Removing the last comment and it should be fine now!!! Thanks for your input!

Reply:
Seems that Travis reports version 4 while on 7?